### PR TITLE
Make `atom-linter-pug` work again

### PR DESCRIPTION
**Pull request by nicolasmn**
The plugin didn't work for me so I fixed it, did the following changes:
- Does not require a config file to work anymore
- Executes `pug-lint` directly instead of using the cli

### CODE REVIEW

**Comment by Nicholaiii** [comment]
@nicolasmn Any chance you can publish this untill the maintainers have time to merge this PR? :)

**Comment by nicolasmn** [comment]
@Nicholaiii You can simply install the fixed version directly from GitHub using `apm install nicolasmn/atom-linter-pug/fix`

**Comment by Arcanemagus** [comment]
For future reference, if something isn't getting handled in the AtomLinter org, bug me on [Atom's Slack](https://atomio.slack.com/) ([invite](http://atom-slack.herokuapp.com/)) 😉.

Looking over this now.

File: lib/init.coffee [hunk 1]
```diff
@@ -59,34 +60,31 @@ module.exports =
...
 
-        parameters = [filePath]
-
-        if !projectConfigPath || !projectConfigPath.configPath
-          if !@onlyRunWhenConfig
+        if !projectConfig || !projectConfig.configPath
+          if @onlyRunWhenConfig
             atom.notifications.addError 'Pug-lint config not found'
-          return Promise.resolve([])
+            return Promise.resolve([])
```
Comment (Arcanemagus) [inline lib/init.coffee]: Shouldn't this _always_ return an empty result if no configuration was found? Currently this change will allow it to continue running when `projectConfig` is empty.

File: lib/init.coffee [hunk 2]
```diff
@@ -59,34 +60,31 @@ module.exports =
...
-              messages.push
-                type: if match[1] then match[1] else 'Error'
-                text: match[5]
-                filePath: match[2]
-                range: helpers.generateRange(textEditor, match[3] - 1, match[4] - 1)
-            return messages
+          resolve results.map (res) -> {
+            type: res.name
+            filePath: filePath
+            range: helpers.generateRange textEditor,  res.line-1, res.column-1
```
Comment (Arcanemagus) [inline lib/init.coffee]: Spaces around the operators please.

File: lib/init.coffee (hunk 1 above)
Comment (nicolasmn) [inline lib/init.coffee]: That's the point, I want `pug-lint` to run on all my .pug files so it shows syntax errors as well. With the current plugin my files won't get linted at all if I don't have a config in every project I use PUG in.

**Comment by nicolasmn** [comment]
@Arcanemagus The CLI gave me unexpected results for syntax errors and I felt like it's more appropriate to use the Node.js API directly.
PS: Sorry for messing up the custom executable path thing. That was pure ignorance, I'll fix it as soon as I have time!

File: lib/init.coffee (hunk 1 above)
Comment (Arcanemagus) [inline lib/init.coffee]: Ah, some linters completely die if they don't have a configuration, or just don't do anything. Others, like `pug-lint` apparently, provide a basic syntax check. I don't use `pug-lint` myself so I didn't know where it fell on that spectrum, only what the existing code was doing 😉.

**Comment by nicolasmn** [comment]
@Arcanemagus The problem with the CLI is that it won't lint unsaved changes even if I just give it the editors text. So, to restore the feature of setting a custom path we would have to resort to the node_module itself instead of the executable which would break the plugin for anyone who had set a custom path and it would make it harder to actually find the correct path since you can't do a simple `where pug-lint` anymore. Thoughts?

**Comment by Arcanemagus** [comment]
@nicolasmn Unfortunately that's the tradeoff of supporting loading the Node.js API from projects, it gets _really_ complex trying to make it work "right" for the user without needing configuration.

You can see how `linter-eslint` does it here:

https://github.com/AtomLinter/linter-eslint/blob/478a4edd21d1277433c85ae7a06ade1c895275ba/src/worker-helpers.js#L43-L77

> break the plugin for anyone who had set a custom path

This would definitely have to be a major version update, you probably should add a check on whether it was set to the default value, and if not send the user an alert that it likely needs changing.

**Comment by Arcanemagus** [comment]
Fixed this up and got it ready for merging @nicolasmn, thanks for getting this started!

**Pull request merged**